modules/renter/contractor/allowance.go

```diff
@@ -43,6 +43,9 @@ func (c *Contractor) SetAllowance(a modules.Allowance) error {
 	if reflect.DeepEqual(a, modules.Allowance{}) {
 		return c.managedCancelAllowance()
 	}
+	if reflect.DeepEqual(a, c.allowance) {
```

Comment:
hmm. At first I thought this was unsafe, because two `types.Currency` values could have different underlying bytes. But it looks like `big.Int` values are always normalized, so that should never occur.